pkg/client/typed/discovery/discovery_client.go

```diff
@@ -110,7 +110,7 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList
 	if err == nil {
 		apiGroup = apiVersionsToAPIGroup(v)
 	}
-	if err != nil && !errors.IsNotFound(err) && !errors.IsForbidden(err) {
```

Comment:
Also, we returned Forbidden for any path that you didn't have access to, regardless of whether it existed or not.